### PR TITLE
Paymill: Mock tests

### PR DESCRIPTION
**Pull request by SagarKarwande**
Added mock test for following methods
- Authorize
- Capture
- Purchase
- Void
- Refund

Refactored gateway ResponseHandler for some edge cases.
Added refund functionality.

## TODO
* [ ] Mock test for save_card
- [x] Support for money protocol
- [x] Add docs and examples

## Edit
1. Many tests are misleading/wrong (for ex: `authorize with invalid cvv`)
2. `ResponseHandler` has a lot of opportunities for refactoring.
3. `save_card` and `save` can probably be used as `store`.
4. The whole token business should be documented in the `moduledoc`

### CODE REVIEW

File: lib/gringotts/gateways/paymill.ex [hunk 1]
```diff
@@ -122,6 +121,13 @@ defmodule Gringotts.Gateways.Paymill do
     commit(:delete, "preauthorizations/#{authorization}", [], options)
   end
 
+  @spec refund(number, String.t, Keyword) :: {:ok | :error, Response}
+  def refund(amount, authorization, options) do
+    post = [amount: amount]
+
```
Comment (kartikjagdale) [inline lib/gringotts/gateways/paymill.ex]: extra line?

File: test/gateways/paymill_test.exs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+      MockResponse.refund_with_high_amount end] do
+
+        {:error, response} = Paymill.refund(400, "tran_d9df8ae460354befe6aee6916fbf", @options)
+
+        refute response.success
+        assert response.status_code == 400
+        assert response.message == "Amount to high"
+      end
+    end
+
```
Comment (kartikjagdale) [inline test/gateways/paymill_test.exs]: extra line?

File: lib/gringotts/gateways/paymill.ex [hunk 3]
```diff
@@ -323,30 +329,62 @@ defmodule Gringotts.Gateways.Paymill do
...
+      |> handle_error(body)
       |> set_params(body)
       |> handle_opts()
     end
+    def parse({:ok, %HTTPoison.Response{body: body, status_code: 403}}) do
+      body = Poison.decode!(body)
+      [status_code: 403, success: false]
+      |> parse_body(body)
+    end
+    def parse(({:ok, %HTTPoison.Response{body: body, status_code: 409}})) do
```
[the remark this answers is not in the document]
Comment (kartikjagdale) [reply lib/gringotts/gateways/paymill.ex]: for functions having similar body and just one argument diff i.e status_code can we do something like 
```
def func_name(status_code) when status_code in [202, 201, 403] do
   similar_body
end
```

**Comment by oyeb** [comment]
Could you summarize this PR and expand the todo (for just this PR)?
